### PR TITLE
Use context instead of cloneElement in RadioBox

### PR DESCRIPTION
**Pull request by jetpacmonkey**
## ✍️ Proposed changes

`RadioBoxGroup` using `cloneElement` to communicate with its child `RadioBox` elements means that those `RadioBox` elements have to be direct children, and it's not possible for consumers to make their own wrapper components. Communicating through context instead removes this limitation and allows the `RadioBox` to be deeper in the React tree.

🎟 _Jira ticket:_ Tangentially related to [CLOUDP-90548](https://jira.mongodb.org/browse/CLOUDP-90548)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## 💬 Further comments

This is as much a discussion-starter as a specific proposed change. I've argued against `cloneElement` for a while, so we don't have to rehash all of that, but I do think that using context allows us to use React more naturally, without having to worry about the additional constraints brought on by a parent component meddling with its children's props.

### CODE REVIEW

File: packages/radio-box-group/src/context.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  onChange: ChangeEventHandler<HTMLInputElement>;
+  getNextId: () => string;
+  size: Size | undefined;
+  name: string | undefined;
+}
+
+const context = createContext<RadioBoxGroupContext | null>(null);
+
+export const Provider = context.Provider;
+export const Consumer = context.Consumer;
```
Comment (brookescarlett) [inline packages/radio-box-group/src/context.ts:15]: Is this being used anywhere? I can't find it 👀

Comment (jetpacmonkey) [reply packages/radio-box-group/src/context.ts:15]: It's not, I added this export before I checked if `RadioBox` was a function or a class component

**Comment by brookescarlett** [comment]
This is looking great to me! @DesignerDave did you want to take a look as well

**Comment by DesignerDave** [comment]
@brookescarlett If it looks good to you, let's ship it 😄

File: packages/radio-box-group/src/context.ts (hunk 1 above)
Comment (brookescarlett) [reply packages/radio-box-group/src/context.ts:15]: We can rm since its not :)

**Review by brookescarlett — APPROVED** [review]

**Comment by jetpacmonkey** [comment]
Thanks for the quick review!

**Pull request merged**